packages/react/src/reactrouter.tsx

```diff
@@ -224,14 +224,15 @@ function computeRootMatch(pathname: string): Match {
 export function withSentryRouting<P extends Record<string, any>, R extends React.ComponentType<P>>(Route: R): R {
   const componentDisplayName = (Route as any).displayName || (Route as any).name;
 
-  const activeRootSpan = getActiveRootSpan();
-
   const WrappedRoute: React.FC<P> = (props: P) => {
     if (props && props.computedMatch && props.computedMatch.isExact) {
-      getCurrentScope().setTransactionName(props.computedMatch.path);
+      const route = props.computedMatch.path;
+      const activeRootSpan = getActiveRootSpan();
```

Comment:
This is the actual fix - we got the root span at instrument time, not at route-render time, leading to this not working as expected 😬

Reply:
That's a tricky one! Good catch 👍🏻